### PR TITLE
Switch to testify/mock and move away from go.hollow.sh/toolbox/events

### PR DESCRIPTION
**Pull request by joelrebel**
#### What does this PR do

- Switches imports to `github.com/metal-toolbox/events` from `go.hollow.sh/toolbox/events`
- Changes mocks to be generated by Mockery which is in use by rivets and (imo) is nicer to work with.
- Updates all tests to use testify/mock instead of gomock.

### CODE REVIEW

**Review by ofaurax — APPROVED** [review]

File: pkg/api/v1/client/client_auth_test.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func newAuthTester(t *testing.T, authToken string) (*integrationTester, finalize
 
 	ctrl := gomock.NewController(t)
```
Comment (DoctorVin) [inline pkg/api/v1/client/client_auth_test.go:39]: We shouldn't need this anymore?

**Review by DoctorVin — CHANGES_REQUESTED** [review]

File: pkg/api/v1/client/client_auth_test.go (hunk 1 above)
Comment (joelrebel) [reply pkg/api/v1/client/client_auth_test.go]: nice catch, thanks, fixed

File: pkg/api/v1/routes/handlers_test.go [hunk 2]
```diff
@@ -393,6 +331,17 @@ func TestAddServer(t *testing.T) {
 			server.ServeHTTP(recorder, tc.request(t))
 
 			tc.assertResponse(t, recorder)
+
+			// Assert all expectations
```
Comment (DoctorVin) [inline pkg/api/v1/routes/handlers_test.go:335]: Why are we doing this here and not in other tests? The GoMock ‘finisher’ asserts the mock expectations.

**Review by DoctorVin — COMMENTED** [review]
I don’t understand the inconsistency in asserting expectations on the mock?

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**